pkg/xds/bootstrap/template_v3.go

```diff
@@ -186,6 +189,56 @@ func genConfig(parameters configParameters, useTokenPath bool) (*envoy_bootstrap
 		}
 	}
 
+	if parameters.IsGatewayDataplane {
+		if maxBytesRaw, ok := parameters.DynamicMetadata["max_heap_size_bytes"]; ok {
+			maxBytes, err := strconv.ParseUint(maxBytesRaw, 10, 64)
+			if err != nil {
+				return nil, err
+			}
+
+			config := &resource_monitors_fixed_heap.FixedHeapConfig{
+				MaxHeapSizeBytes: maxBytes,
+			}
+			marshaledConfig, err := util_proto.MarshalAnyDeterministic(config)
+			if err != nil {
+				return nil, errors.Wrapf(err, "could not marshall %T", config)
+			}
+
+			fixedHeap := "envoy.resource_monitors.fixed_heap"
+
+			res.OverloadManager = &envoy_overload_v3.OverloadManager{
```

Comment:
Random question but should this be actually not only limited to gateway but applies to every dp?

Reply:
Probably? But I think that's a bit of a can of worms because there's no one size fits all for inside your mesh. In the role of an edge proxy we can be quite strict, plus we have the Envoy docs to give us a sensible default for this case.

Reply:
Ok let's not scope creep this issue :) Can you open a follow up issue about generalising this?

Reply:
No yeah definitely.